app.py

```diff
@@ -17,18 +17,37 @@ def home() -> str:
     return render_template('home.html')
 
 
+@app.route('/health', methods=['GET'])
+def health_check() -> tuple[str, int]:
+    return 'OK', 200
+
+
 @app.route('/animals', methods=['GET'])
 def index() -> Response:
-    animals = Animal.query.all()
-    return jsonify({"animals": [AnimalResponse.model_validate(animal).model_dump(mode='json') for animal in animals]})
+    name = request.args.get('name')
+    if name:
+        animals = Animal.query.filter_by(name=name).all()
```

Comment:
You can improve it to search even by substring
https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.ColumnOperators.contains

Reply:
@marcustas fixed - 
<img width="1377" alt="Screenshot 2023-10-15 at 12 34 20" src="https://github.com/marcustas/course_flask/assets/69207522/3d286fd1-28cf-4111-beb3-678dbfe3ea42">